pkg/apis/machine/v1alpha1/shared_types.go

```diff
@@ -72,3 +72,15 @@ type MachineSummary struct {
 	// OwnerRef
 	OwnerRef string `json:"ownerRef,omitempty"`
 }
+
+const (
+	// ClusterAutoscalerScaleDownDisabledAnnotationKey annotation to disable the scale-down of the nodes.
+	ClusterAutoscalerScaleDownDisabledAnnotationKey = "cluster-autoscaler.kubernetes.io/scale-down-disabled"
+	// ClusterAutoscalerScaleDownDisabledAnnotationValue annotation to disable the scale-down of the nodes.
+	ClusterAutoscalerScaleDownDisabledAnnotationValue = "True"
+
+	// ClusterAutoscalerScaleDownDisabledAnnotationByMCMKey annotation to disable the scale-down of the nodes.
+	ClusterAutoscalerScaleDownDisabledAnnotationByMCMKey = "cluster-autoscaler.kubernetes.io/scale-down-disabled-by-mcm"
+	// ClusterAutoscalerScaleDownDisabledAnnotationByMCMValue annotation to disable the scale-down of the nodes.
+	ClusterAutoscalerScaleDownDisabledAnnotationByMCMValue = "True"
+)
```

Comment:
I would expect that these consts to be moved out of this package as it's less related to the APIs. 
Probably `pkg/util/*` or `pkg/controller` ?

Reply:
Hm, I kept it here, thinking this keyword will actually be an API contract for external consumption. eg Cluster autoscaler. This has to be maintained across implementations, though open for the thoughts.

Reply:
Okay. I would prefer to keep it outside the `pkg/apis/machines` as it has very little to do with actual machine CRD contracts. Also, this code in this pkg should be duplicated then to even - https://github.com/gardener/machine-controller-manager/blob/44d70a27e902b52d98d44da7717bc4ad471d099d/pkg/apis/machine/v1alpha1/shared_types.go. I honestly don't think we should be adding such consts to these files which are used for auto generation and are duplicated. 

However, i do get your point about the API contracts part. How about perhaps moving this to something like `pkg/apis/clusterautoscaler`?

Reply:
Good point, I moved it to the utils file of deployment with this [change](https://github.com/gardener/machine-controller-manager/pull/496/commits/3a557bbf6fc38ef5ade63b869364dff462d18558).

Reply:
Thank you.